### PR TITLE
THEMES-714 - Convert section title block to v2 and update tests to RTL

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Convert section title block to v2 and update tests to use RTL

## Jira Ticket

- [THEMES-714](https://arcpublishing.atlassian.net/browse/THEMES-714)

## Test Steps

- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout THEMES-714`
2. Checkout out feature pack branch `git checkout THEMES-714`
3. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/section-title-block`
4. Add Section title block to page and config block to use `site-service-hierarchy` content source with `hierarchy` value as `default`
5. Validate section title and links display
6. Now remove the content source values and toggle the `inheritGlobalContent` custom field and set global content to use the same content source as step 4
7. Validate section title and links display

## Dependencies or Side Effects

Feature Pack PR - https://github.com/WPMedia/arc-themes-feature-pack/pull/366
Manifest PR - https://github.com/WPMedia/arc-themes-manifests/pull/151

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/section-title-block/features/section-title/default.jsx [hunk 1]
```diff
@@ -1,16 +1,55 @@
...
+							}
+							return null;
+						})}
+				</Join>
+			</Stack>
+		</Stack>
+	) : null;
+
+const SectionTitleContainer = ({ customFields }) => {
+	const { inheritGlobalContent = true, sectionContentConfig } = customFields;
```
Comment (vgalatro) [inline blocks/section-title-block/features/section-title/default.jsx:42]: The related custom field in the prop types below should have a `defaultValue: true` as well, so the editor visual matches what the code is actually doing.

**Review by vgalatro — DISMISSED** [review]
Just a small change needed, to have consistency between the editor UI and programmatic defaults.

File: blocks/section-title-block/features/section-title/default.jsx (hunk 1 above)
Comment (matthewroach) [reply blocks/section-title-block/features/section-title/default.jsx:42]: @vgalatro Fixed

**Review by vgalatro — DISMISSED** [review]
Changes look good, approved.

**Pull request merged**